vent/controller/control_module.py

```diff
@@ -347,8 +348,9 @@ def get_sensors(self):
                                           vte=self.vte,
                                           breaths_per_minute=self.bpm,
                                           inspiration_time_sec=self.I_phase,
-                                          timestamp=time.time())
-
+                                          timestamp=time.time(),
+                                          loop_counter = self.loop_counter)
```

Comment:
These variables (updated by main_loop) are not thread safe. So you might need to add a lock when access them. Also rename them to shard_xxx

Reply:
added a lock. I'm not entirely sure though, which variables are shared and which are not.